assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,502 @@
+####         the `R1` files in `fastq` (after decompressing them into text files),
+####         and also count up all the lines in the `R1` files in `fastq`. 
+####         Note they should be the same.
+
+(base) [002-unix-intro]--% zcat fastq/*.R1.fq.gz | wc
+  19364   19364 1678394
+
+(base) [002-unix-intro]--% mkdir R1_unzipped
+(base) [002-unix-intro]--% cp fastq/*R1.fq.gz R1_unzipped/
+(base) [002-unix-intro]--% cd R1_unzipped/
+(base) [R1_unzipped]--% gunzip *.fq.gz
+(base) [R1_unzipped]--% cd ..
+(base) [002-unix-intro]--% cat R1_unzipped/*.fq | wc
+  19364   19364 1678394
+
+#### Ex. 37: Now, gzcat all the `R1` files in `fastq` and redirect that
+####         to a file called `R1_all_via_gzcat.fq` in the top directory of the repo
+
+(base) [002-unix-intro]--% zcat fastq/*.R1.fq.gz > ../../R1_all_via_gzcat.fq
+
+#### Ex. 38: People are usually familiar with using `cat` to catenate text
+####         files. It turns out that you can also catenate `gzip` compressed files. 
+####         The result is another properly gzipped file that can be decompressed. 
+####         Catenate all the gzipped `R1` files in `fastq` into a single (still gzipped)
+####         file called `R1_all_via_cat.fq.gz`
+
+(base) [002-unix-intro]--% cat fastq/*.R1.fq.gz > R1_all_via_cat.fq.gz
+
+#### Ex. 39: Copy `R1_all_via_cat.fq.gz` to `copy_of_R1_all_via_cat.fq.gz`
+
+(base) [002-unix-intro]--% cp R1_all_via_cat.fq.gz copy_of_R1_all_via_cat.fq.gz
+
+#### Ex. 40: Now, decompress `R1_all_via_cat.fq.gz` into `R1_all_via_cat.fq`
+
+(base) [002-unix-intro]--% gunzip R1_all_via_cat.fq.gz 
+
+#### Ex. 41: Compute the SHA1 hashes of `R1_all_via_cat.fq` and `R1_all_via_gzcat.fq`
+####         to confirm they are identical
+
+(base) [002-unix-intro]--% shasum R1_all_via_cat.fq 
+d5b95bf950e8150c56aed90eea9febeb23ddd856  R1_all_via_cat.fq
+
+(base) [002-unix-intro]--% shasum ../../R1_all_via_gzcat.fq 
+d5b95bf950e8150c56aed90eea9febeb23ddd856  ../../R1_all_via_gzcat.fq
+
+#### Ex. 42: But now, gzip `R1_all_via_gzcat.fq` and compare the result to
+####         `copy_of_R1_all_via_cat.fq.gz` by computing the SHA1 hash of each
+
+(base) [002-unix-intro]--% gzip R1_all_via_cat.fq 
+(base) [002-unix-intro]--% shasum R1_all_via_cat.fq.gz 
+56a36f0a4ff86a378252aed18a394e57932e0318  R1_all_via_cat.fq.gz
+
+(base) [002-unix-intro]--% shasum copy_of_R1_all_via_cat.fq.gz 
+93588c00857e42325bf64bb0ea5992307460737c  copy_of_R1_all_via_cat.fq.gz
+
+#### Ex. 43: using `du` with the `-h` (i.e. "human readable" option) print the
+####         file sizes of `R1_all_via_cat.fq` and `copy_of_R1_all_via_cat.fq.gz`.
+
+(base) [002-unix-intro]--% du -h R1_all_via_cat.fq copy_of_R1_all_via_cat.fq.gz 
+512	R1_all_via_cat.fq
+412K	copy_of_R1_all_via_cat.fq.gz
+
+(base) [002-unix-intro]--% ls -l R1_all_via_cat.fq copy_of_R1_all_via_cat.fq.gz 
+-rw-r--r-- 1 emcampbe 418K Jan 25 15:56 copy_of_R1_all_via_cat.fq.gz
+-rw-r--r-- 1 emcampbe 1.7M Jan 25 16:02 R1_all_via_cat.fq
+
```

Comment:
That is weird. I don't know why du and ls are giving such different results.  :-/